File: 프로그래머스/2/42626. 더 맵게/더 맵게.py
import heapq


def solution(scoville, K):
    answer = 0
    heapq.heapify(scoville)
    while len(scoville)>=1:
        a = heapq.heappop(scoville)
        if a>=K:
            break
        if len(scoville)>=1:
            b = heapq.heappop(scoville)
        else:
            return -1
        c = a + (b*2)
        heapq.heappush(scoville, c)
        answer+=1
    return answer

File: 프로그래머스/2/42626. 더 맵게/더 맵게.java
import java.util.PriorityQueue;

class Solution {
    public int solution(int[] scoville, int K) {
        int answer = 0;
        PriorityQueue<Integer> heap = new PriorityQueue<>();
        for (int s: scoville) {
            heap.offer(s);
        }
        while (!heap.isEmpty()) {
            int a = heap.poll();
            if (a>=K) {
                break;
            }
            if (heap.isEmpty()) {
                return -1;
            }
            int b = heap.poll();
            int c = a + (b*2);
            heap.offer(c);
            answer++;
        }
        return answer;
    }
}